File: app/src/main/java/com/playground/ui/noteslist/NoteListViewModel.java
package com.playground.ui.noteslist;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;
import android.arch.paging.PagedList;

import com.playground.data.RepositoryNotes;
import com.playground.entities.NoteEntry;

/**
 * Created by emil.ivanov on 9/9/18.
 */
public class NoteListViewModel extends ViewModel {

    RepositoryNotes repository;

    public NoteListViewModel(RepositoryNotes repositoryNotes) {
        this.repository = repositoryNotes;
    }

    public LiveData<PagedList<NoteEntry>> getNotes() {
        return repository.loadNotes().getData();
    }

    public void delete(NoteEntry noteEntry) {
        repository.deleteNote(noteEntry);
    }

}

File: app/src/mainKotlin/kotlin/com.playground/ui/noteslist/NoteListViewModel.kt
package com.playground.ui.noteslist

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel
import android.arch.paging.PagedList
import com.playground.data.RepositoryNotes
import com.playground.entities.NoteEntry

/**
 * Created by emil.ivanov on 9/9/18.
 */
class NoteListViewModel(val repository: RepositoryNotes) : ViewModel() {
    fun getNotes(): LiveData<PagedList<NoteEntry>> {
        return repository.loadNotes().data
    }

    fun delete(noteEntry: NoteEntry) {
        repository.deleteNote(noteEntry)
    }

}
